In [5]:
from sympy import *

a, b, l = symbols('a b lambda')

Целевая функция для аргументов a и b

In [6]:
g = 3 * a**2 + a + 4 * b + 5 * b**2
g

3*a**2 + a + 5*b**2 + 4*b

Функция ограничений

In [7]:
q = a + b - 200
q

a + b - 200

Функция Лагранжа

In [8]:
f = g + l * q
f

3*a**2 + a + 5*b**2 + 4*b + lambda*(a + b - 200)

Расчет первых производных

In [17]:
f_diff_a = f.diff(a)
f_diff_b = f.diff(b)
f_diff_l = f.diff(l) # первая производная по лямбде всегда равна просто функции ограничений, т.е. ее можно не считать

Приравняю первые производные к нулю и решу систему уравнений

In [13]:
# можно так, не приравнивая явно
# sols = solve([f_diff_a, f_diff_b, f_diff_l], a, b, l)
# sols

In [14]:
eq1 = Eq(f_diff_a, 0)
eq2 = Eq(f_diff_b, 0)
eq3 = Eq(f_diff_l, 0)
 
sol = solve([eq1, eq2, eq3], [a, b, l])
sol

{a: 2003/16, b: 1197/16, lambda: -6017/8}

In [16]:
print(f'Стационарная точка M(x, y): {float(sol[a])}, {float(sol[b])}')

Стационарная точка M(x, y): 125.1875, 74.8125


Т.е. на станке a надо произвести 125 упаковок, а на станке b 75 упаковок

In [20]:
L, W, H, l = symbols('L W H lambda')

In [22]:
f = -2 * (L*W + W*H + H*L) + l * (L*W*H - 1)
f

-2*H*L - 2*H*W - 2*L*W + lambda*(H*L*W - 1)

In [25]:
f_diff_l = f.diff(L)
f_diff_w = f.diff(W)
f_diff_h = f.diff(H)
f_diff_l = f.diff(l)

In [26]:
eq1 = Eq(f_diff_l, 0)
eq2 = Eq(f_diff_w, 0)
eq3 = Eq(f_diff_h, 0)
eq4 = Eq(f_diff_l, 0)
 
sol = solve([eq1, eq2, eq3, eq4], [L, W, H, l])
sol

[(W**(-2), W, W, 2*(W**3 + 1)/W)]

In [ ]:
-2 * (L*W + W*H + H*L)

In [30]:
x = 4**1/3

In [31]:
2 * ( x*x + x*x + x*x)

10.666666666666666

In [32]:
# Импортируем все необходимые функции из библиотеки SymPy
from sympy import init_printing, solve, symbols

# Подключаем аккуратный вывод формул
init_printing()

# Создаём переменные
x1, x2, x1_tilde, lambda1, lambda2 = symbols(
    "x_1 x_2 xtilde_2 \lambda_1 \lambda_2", real=True
)

# Пишем систему уравнений для переменных
dLdx1 = -x2 + lambda1 - lambda2
dLdx2 = -x1 + lambda1
dLdx1_tilde = 2 * lambda2 * x1_tilde
dLdlambda1 = x1 + x2 - 10
dLdlambda2 = -x1 + 6 + x1_tilde**2

# Решаем систему уравнений с помощью функции solve
solution = solve(
    [dLdx1, dLdx2, dLdx1_tilde, dLdlambda1, dLdlambda2],
    x1,
    x2,
    x1_tilde,
    lambda1,
    lambda2,
)

print(solution)

[(6, 4, 0, 6, 2)]
